src/NUnitFramework/framework/Internal/Builders/DefaultTestCaseBuilder.cs

```diff
@@ -141,7 +142,7 @@ public Test BuildFrom(IMethodInfo method, Test parentSuite)
                     tests.Add(test);
             }
 
-            return tests.Count > 0
+            return tests.Count > 0 || builders.OfType<TestCaseSourceAttribute>().Any()
```

Comment:
@CharliePoole Lots of failures at [`TestBuilder.MakeTestCase`'s  `Assert.That(test, Is.TypeOf<TestMethod>());`](https://github.com/nunit/nunit/blob/master/src/NUnitFramework/TestBuilder.cs#L82) because it's actually getting a `ParameterizedMethodSuite`.